applications/StructuralMechanicsApplication/custom_elements/truss_element_linear_3D2N.cpp

```diff
@@ -261,6 +261,13 @@ void TrussElementLinear3D2N::FinalizeSolutionStep(const ProcessInfo& rCurrentPro
     KRATOS_CATCH("");
 }
 
+double TrussElementLinear3D2N::ReturnTangentModulus1D(const ProcessInfo& rCurrentProcessInfo)

```

Comment:
these methods can be `const` actually, just saying :)

Reply:
Hi @AlejandroCornejo, thank you very much for your comment. The sad thing is, although I fully agree with the intent of your suggestion, it isn't easy to implement. If you look at the `override` of `ReturnTangentModulus1D` in class `TrussElementLinear3D2N` you will see it calls member function `CalculateLinearStrain`, which is not `const`. To make it `const`, we would need to also make `CreateTransformationMatrix` `const` (since it isn't at present). If you dive deeper, you'll find that `CreateTransformationMatrix` also depends on non-`const` members (e.g. `WriteTransformationCoordinates`), some of them being `virtual`, which means that derived classes may `override` them (and thus also need to be changed). In other words, although I value your suggestion and it may seem simple at first sight, reality is that it requires a substantial amount of work to implement. If you insist, I will try to make all required members `const`, but I feel that the required changes are beyond the scope of this PR and they're better addressed by a separate PR. Please let me know what you prefer. Thank you.

Reply:
yes yes of course I understand!! if at some point it has to be done, it will imply a lot of refactoring...